Parsers/ASimAuthentication/Parsers/vimAuthenticationGithub.yaml

```diff
@@ -0,0 +1,133 @@
+     srcipaddr_has_any_prefix:dynamic = dynamic([]),
+     srchostname_has_any:dynamic = dynamic([]),
+     eventtype_in:dynamic =dynamic([]),
+     eventresultdetails_in:dynamic = dynamic([]),
+     eventresult:string = '*',
+     disabled:bool=false
+     )
+     {
+     let EventResultDetailslookup=datatable(EventOriginalResultDetails:string,EventResult:string,EventType:string,EventSeverity:string )
+     ["user.login","Success","logon","Informational",
+     "user.logout","Success","logoff","Informational",
+     "user.two_factor_challenge_success","Success","logon","Informational",
+     "user.two_factor_challenge_failure","Failure","logon","Informational",
+     "user.two_factor_requested","Success","logon","Informational",
+     "user.failed_login","Failure","logon","Informational",
+     "user.sign_in_from_unrecognized_device_and_location","Success","logon","High",
+     "user.sign_in_from_unrecognized_device","NA","logon","High",
+     "user.reset_password","Success","logon","Informational",
+     "user.device_verification_requested","Success","logon","Informational",
+     "user.device_verification_requested","Success","logon","Informational",
+     "user.two_factor_recovery_codes_downloaded","Success","logon","Informational",
+     "user.forgot_password","Success","logon","Informational",
+     "user.new_device_used","Success","logon","Informational",
+     "user.promote","Success","Elevate","Informational",
+     ];
+     let GithubAuthenticationEvent=EventResultDetailslookup 
+     |project EventOriginalResultDetails
+     | summarize make_set(EventOriginalResultDetails);r.promote","Success","Elevate","Informational",
+     ];
+     GitHubAuditLogPolling_CL
+     |where not(disabled)
+     | extend TimeGenerated = todatetime(tostring(split(TimeGenerated, '.', 0)[0]))      
+     |where operation_type_s =='authentication'
+     |project TimeGenerated=unixtime_milliseconds_todatetime(created_at_d),   
+        EventCount=int(1),
+        EventProduct = 'Github',
+        EventVendor = 'Microsoft',
+        EventSchema='Authentication',
+        EventSchemaVersion='0.1.3',
+        EventUid =_ItemId,
+        TargetAppname='Github',
+        TargetAppType='SaaS application'
+        EventOriginalUid= _document_id_s,
+        TargetUsername=actor_s,
+        UserId=user_id_d,
+        SrcIpAddress=actor_ips_s,
+        SrcGeoCountry=actor_location_country_code_s,
+        EventStartTime=TimeGenerated,
+        EventEndtime=TimeGenerated,
+        EventOriginalResultDetails=action_s    
+     |lookup EventResultDetailslookup on EventOriginalResultDetails    
+     | where 
+         (isnull(starttime) or TimeGenerated >= starttime) 
+          and (isnull(endtime) or TimeGenerated <= endtime)
+          and ((array_length(username_has_any) == 0) or (TargetUsername  has_any (username_has_any)))
+          and ((array_length(targetappname_has_any) == 0)) // targetappname not available on source
```

Comment:
we can use static value check like :
and ((array_length(targetappname_has_any) == 0) or 'Github' has_any targetappname_has_any)